service/src/main/java/io/camunda/service/entities/ProcessInstanceEntity.java

```diff
@@ -8,14 +8,15 @@
 package io.camunda.service.entities;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.camunda.service.security.auth.ResourceAuthorizationKey;
 import java.util.List;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public record ProcessInstanceEntity(
     Long key,
     String processName,
     Integer processVersion,
-    String bpmnProcessId,
+    @ResourceAuthorizationKey(forResourceType = "process-definition") String bpmnProcessId,
```

Comment:
🔧  I like the approach here, I would instead rely on the `AuthorizationResourceType` class to give you the resource type that maps to the authorizations

Reply:
yes, good point, I was about to create a new enum 😄 
I wonder why `AuthorizationResourceType` is not used in `AuthorizationEntity` ? 🤔

Reply:
Good question, I think thats just an oversight on my part to be honest and something that I should update

Reply:
Hm... it implies that the field name is also called `bpmnProcessId,` which might be okay in that case, but I could imagine that other fields are named differently depending on the index.